File: Easy/Two Repeated Elements.cs
//{ Driver Code Starts
//Initial Template for C#

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {

                int n = Convert.ToInt32(Console.ReadLine());
                int[] arr = new int[n+2];
                var stringArray = Console.ReadLine().Split(' ');
                int j = 0;
                for (int i = 0; i < stringArray.Length; i++)
                {

                    if (stringArray[i].CompareTo(" ") != -1)
                    {
                        arr[j] = int.Parse(stringArray[i]);
                        j++;
                    }
                }
                Solution obj = new Solution();
                var res = obj.twoRepeated(arr,n);
                Console.WriteLine(res[0] + " " + res[1]);
            }

        }
    }

    
// } Driver Code Ends
//User function Template for C#

    class Solution
    {
       //Function to find two repeated elements.
        public List<int> twoRepeated(int[] a,int n)
        {
            //code here
            var ans = new List<int>();
            
            var mappa = new HashSet<int>();
            
            for (int i=0; i<n+2; i++)
            {
                if (mappa.Contains(a[i])) ans.Add(a[i]);
                else mappa.Add(a[i]);
            }
            
            return ans;
        }

    }

}

//{ Driver Code Starts.
    
// } Driver Code Ends

File: Medium/Two Repeated Elements.cpp
//{ Driver Code Starts
//Initial template for C++

#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++

class Solution
{
  public:
    //Function to find two repeated elements.
    vector<int> twoRepeated (int arr[], int n)
    {
        // Your code here
        n += 2;
        map<int, int> m;
        vector<int> ans;

        for (int i = 0; i < n; i++)
        {
            if (m.find(arr[i]) == m.end())
            {
                m[arr[i]] = 1;
            }
            else
            {
                m[arr[i]]++;
                ans.push_back(arr[i]);
            }
        }
        
        return ans;
    }
};

//{ Driver Code Starts.

int main()
{
    int t,n;
    cin>>t;
    
    while(t--)
    {
        cin>>n;
        
        int a[n+2];
        
        for(int i=0;i<n+2;i++)
            cin>>a[i];
        
        Solution obj;
        vector<int> res;
        res = obj.twoRepeated(a, n);
        cout<<res[0]<<" "<<res[1]<<endl;
    }
    return 0;
}

// } Driver Code Ends
